src/Bicep.Core/TypeSystem/IResourceTypeProvider.cs

```diff
@@ -26,5 +26,10 @@ public interface IResourceTypeProvider
         /// Returns the full list of available types defined by this provider.
         /// </summary>
         IEnumerable<ResourceTypeReference> GetAvailableTypes();
+
+        /// <summary>
+        /// Returns the version of the type definitions used by this provider.
```

Comment:
Is there an expected format?  If so, it should be documented.  Are we assuming semver?

Reply:
for now its the legacy format, we need to design this better in the future. We have not decided on versioning yet. I have thoughts about it and think it should be a data (similar to api-versions) but for now its whatever was being used before.
Its a TODO for sure to agree on the versioning scheme but its out of scope of this PR IMO. Resolving for now but ACK that this needs more TLC